exoscale/loadbalancer.go

```diff
@@ -239,14 +240,23 @@ next:
 			// If a service exposing the same port already exists,
 			// flag it for update and save its ID for later reference.
 			if nlbServiceUpdate.Port == nlbServiceCurrent.Port {
-				debugf("service port %d already in use by NLB service %q, marking for update",
+				debugf("Service port %d already in use by NLB service %s/%s, marking for update",
 					nlbServiceCurrent.Port,
+					nlbCurrent.Name,
 					nlbServiceCurrent.Name)
 				nlbServices[nlbServiceCurrent.Port] = nlbServiceCurrent
 				continue next
 			}
 		}
 
+		if l.isExternal(service) {
+			debugf("NLB service %s/%s doesn't match any service port, but this Service is "+
+				"using an external NLB so we don't delete it as it may belong to another Service",
```

Comment:
```suggestion
			debugf("NLB service %s/%s doesn't match any service port, but this Service is "+
				"using an external NLB. Avoiding deletion since it may belong to another Service",
```